encodings/decimal-byte-parts/src/decimal_byte_parts/mod.rs

```diff
@@ -82,27 +80,6 @@ impl DecimalBytePartsArray {
             vortex_bail!("decimal bytes parts 2nd to 4th must be non-nullable u64 primitive typed")
         }
 
-        let primitive_bit_width = iter::once(&msp)
-            .chain(&lower_parts)
-            .map(|a| {
-                PType::try_from(a.dtype())
-                    .vortex_expect("already checked")
-                    .bit_width()
-            })
-            .sum();
-
-        if decimal_dtype.required_bit_width() > primitive_bit_width {
-            vortex_bail!(
-                "cannot represent a decimal {decimal_dtype} as primitive parts {:?}, decimal bit width {}, primitive bit width {}",
-                iter::once(&msp)
-                    .chain(&lower_parts)
-                    .map(|a| a.dtype())
-                    .collect_vec(),
-                decimal_dtype.required_bit_width(),
-                primitive_bit_width
-            )
-        }
```

Comment:
This causes errors when you're just holding values of a narrowed integer type

Reply:
What validation will you add instead?